ckan/views/group.py

```diff
@@ -0,0 +1,1160 @@
+        'with_private': False
+    }
+
+    try:
+        _check_access('site_read', context)
+        _check_access('group_list', context)
+    except NotAuthorized:
+        base.abort(403, _('Not authorized to see this page'))
+
+    q = c.q = request.params.get('q', '')
+    sort_by = c.sort_by_selected = request.params.get('sort')
+
+    # pass user info to context as needed to view private datasets of
+    # orgs correctly
+    if c.userobj:
+        context['user_id'] = c.userobj.id
+        context['user_is_admin'] = c.userobj.sysadmin
+
+    try:
+        data_dict_global_results = {
+            'all_fields': False,
+            'q': q,
+            'sort': sort_by,
+            'type': group_type or 'group',
+        }
+        global_results = _action('group_list')(context,
+                                               data_dict_global_results)
+    except ValidationError as e:
+        if e.error_dict and e.error_dict.get('message'):
+            msg = e.error_dict['message']
+        else:
+            msg = str(e)
+        h.flash_error(msg)
+        c.page = h.Page([], 0)
+        return base.render(
+            _index_template(group_type), extra_vars={'group_type': group_type})
+
+    data_dict_page_results = {
+        'all_fields': True,
+        'q': q,
+        'sort': sort_by,
+        'type': group_type or 'group',
+        'limit': items_per_page,
+        'offset': items_per_page * (page - 1),
+        'include_extras': True
+    }
+    page_results = _action('group_list')(context, data_dict_page_results)
+
+    c.page = h.Page(
+        collection=global_results,
+        page=page,
+        url=h.pager_url,
+        items_per_page=items_per_page, )
+
+    c.page.items = page_results
+    vars = dict(group_type=group_type)
+    return base.render(_index_template(group_type), extra_vars=vars)
+
+
+def _read(id, limit, group_type):
+    ''' This is common code used by both read and bulk_process'''
+    context = {
+        'model': model,
+        'session': model.Session,
+        'user': c.user,
+        'schema': _db_to_form_schema(group_type=group_type),
+        'for_view': True,
+        'extras_as_string': True
+    }
+
+    q = c.q = request.params.get('q', '')
+    # Search within group
+    if c.group_dict.get('is_organization'):
+        q += ' owner_org:"%s"' % c.group_dict.get('id')
+    else:
+        q += ' groups:"%s"' % c.group_dict.get('name')
+
+    c.description_formatted = \
+        h.render_markdown(c.group_dict.get('description'))
+
+    context['return_query'] = True
+
+    page = h.get_page_number(request.params)
+
+    # most search operations should reset the page counter:
+    params_nopage = [(k, v) for k, v in request.params.items() if k != 'page']
+    sort_by = request.params.get('sort', None)
+
+    def search_url(params):
+        controller = lookup_group_controller(group_type)
+        action = u'bulk_process' if getattr(c, u'action') == u'bulk_process' else 'read'
```

Comment:
Shouldn't it be `request.blueprint`? without group_controller, there will be no c.action